ui/components/app/signature-request/signature-request-data/signature-request-data.js

```diff
@@ -19,61 +19,68 @@ import {
-            color={COLORS.TEXT_DEFAULT}
-            marginLeft={4}
-            fontWeight={
-              typeof value === 'object' ? FONT_WEIGHT.BOLD : FONT_WEIGHT.NORMAL
+      {Object.entries(data).map((args, i) => {
+        const label = sanitizeString(args[0]);
+        const value = sanitizeString(args[1]);
+        return (
+          <Box
+            className="signature-request-data__node"
+            key={`${label}-${i}`}
+            paddingLeft={2}
+            display={
+              typeof value !== 'object' || value === null ? DISPLAY.FLEX : null
             }
           >
-            {label.charAt(0).toUpperCase() + label.slice(1)}:{' '}
-          </Typography>
-          {typeof value === 'object' && value !== null ? (
-            <SignatureRequestData data={value} />
-          ) : (
             <Typography
               as="span"
               color={COLORS.TEXT_DEFAULT}
               marginLeft={4}
-              className="signature-request-data__node__value"
+              fontWeight={
+                typeof value === 'object'
+                  ? FONT_WEIGHT.BOLD
+                  : FONT_WEIGHT.NORMAL
+              }
             >
-              {isValidHexAddress(value, {
-                mixedCaseUseChecksum: true,
-              }) ? (
-                <Typography
-                  variant={TYPOGRAPHY.H7}
-                  color={COLORS.INFO_DEFAULT}
-                  className="signature-request-data__node__value__address"
-                >
-                  <Address
-                    addressOnly
-                    checksummedRecipientAddress={toChecksumHexAddress(value)}
-                    recipientName={getAccountName(identities, value)}
-                  />
-                </Typography>
-              ) : (
-                `${value}`
-              )}
+              {label.charAt(0).toUpperCase() + label.slice(1)}:{' '}
             </Typography>
-          )}
-        </Box>
-      ))}
+            {typeof value === 'object' && value !== null ? (
+              <SignatureRequestData data={value} />
+            ) : (
+              <Typography
+                as="span"
+                color={COLORS.TEXT_DEFAULT}
+                marginLeft={4}
+                className="signature-request-data__node__value"
+              >
+                {isValidHexAddress(value, {
+                  mixedCaseUseChecksum: true,
+                }) ? (
+                  <Typography
+                    variant={TYPOGRAPHY.H7}
+                    color={COLORS.INFO_DEFAULT}
+                    className="signature-request-data__node__value__address"
+                  >
+                    <Address
+                      addressOnly
+                      checksummedRecipientAddress={toChecksumHexAddress(value)}
+                      recipientName={getAccountName(identities, value)}
+                    />
+                  </Typography>
+                ) : (
+                  value
```

Comment:
```suggestion
                  `${value}`
```

I think we should add the string conversation back. 

If the value is an object, we may get the error `Uncaught Error: Objects are not valid as a React child` here; or if this is a boolean, the value may not print.